components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -111,18 +143,32 @@ export function WorkspaceEntry({ desc, model, isAdmin, stopWorkspace }: Props) {
             </Tooltip>
         </ItemField>
         <ItemFieldContextMenu menuEntries={menuEntries} />
-        {isModalVisible && <ConfirmationModal
+        {isDeleteModalVisible && <ConfirmationModal
             title="Delete Workspace"
             areYouSureText="Are you sure you want to delete this workspace?"
             children={{
                 name: ws.id,
                 description: ws.description,
             }}
             buttonText="Delete Workspace"
-            visible={isModalVisible}
-            onClose={() => setModalVisible(false)}
+            visible={isDeleteModalVisible}
+            onClose={() => setDeleteModalVisible(false)}
             onConfirm={() => model.deleteWorkspace(ws.id)}
         />}
+        <Modal visible={isRenameModalVisible} onClose={() => setRenameModalVisible(false)} onEnter={() => { updateWorkspaceDescription(); return true }}>
+            <h3 className="mb-4">Rename Workspace Description</h3>
+            <div className="border-t border-b border-gray-200 dark:border-gray-800 -mx-6 px-6 py-4 space-y-2">
+                <input className="w-full" type="text" defaultValue={workspaceDescription} ref={renameInputRef} />
```

Comment:
**suggestion**: Could we enforce a character limit let's say of 250 characters as a) it does not make sense to allow more characters since these won't be visible in the dashboard and b) the generic alert dialog when saving with a long description does not provide any useful context, right?

| Long description name hidden in the dashboard | Long description name error |
|-|-|
| <img width="580" alt="Screenshot 2021-09-15 at 7 28 59 PM" src="https://user-images.githubusercontent.com/120486/133472736-3a3a68d1-750f-432a-bd47-211abf16391b.png"> | <img width="463" alt="Screenshot 2021-09-15 at 7 28 34 PM" src="https://user-images.githubusercontent.com/120486/133472742-7def30bf-a2cc-4b73-9077-fa3f153fcefd.png"> |